packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/relay.ts

```diff
@@ -25,8 +25,19 @@ const connections = config.solanaEndpoints.map(
   (endpoint) => new Connection(endpoint)
 )
 
-const delay = async (ms: number) => {
-  return await new Promise((resolve) => setTimeout(resolve, ms))
+const delay = async (ms: number, options?: { signal: AbortSignal }) => {
+  const signal = options?.signal
+  return new Promise<void>((resolve, reject) => {
+    const listener = () => {
```

Comment:
for completeness, I think you may want this
```
if (signal.aborted) {
    reject(signal.reason);
}
```
to avoid a race

Reply:
you mean in the happy path to prevent it from resolving?

Reply:
oh gotcha, like above the listener setup 👍